zcash_primitives/src/transaction/txid.rs

```diff
@@ -531,6 +589,11 @@ impl TransactionDigest<Authorized> for BlockTxCommitmentDigester {
             h.write_all(digest.as_bytes()).unwrap();
         }
 
+        #[cfg(zcash_unstable = "nu6")] /* TODO nu7 */
+        if TxVersion::suggested_for_branch(consensus_branch_id).has_zsa() {
```

Comment:
this looks very error prone. If we have the digest, let's write it.

Reply:
That's how they do it with zfuture, but yeah, we can  do a little better, I guess

Probably need to make them option then